### PR TITLE
Fixed broken links - 2023.10

### PR DESCRIPTION
**Pull request by n-boshnakov**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR fixes and broken links in the public website. It also removes links pointing to outdated and removed content.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc developer
NONE
```

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -43,7 +43,7 @@
 * [`NodeLocalDNS` feature](usage/node-local-dns.md)
```
Comment (vpnachev) [inline docs/README.md:43]: While you are on this file, there is a link https://www.sap-tv.com/video/40962/gardener-planting-the-seeds-of-success-in-the-cloud to a resource that seems to be no longer existing, can you please take a look?


EDIT: The broken link is in the README.md file in the root directory, not in the docs/ folder. Excuse me for the misunderstanding.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**